ITE: drivers/bbram: add magic number to compare in initial

GTLin08 opened the pull request:
BBRAM content should be cleared when boot from cutoff. Compare magic
number to decide whether to clear entire BBRAM.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

GTLin08 committed:
ITE: drivers/bbram: add magic number to compare in initial

BBRAM content should be cleared when boot from cutoff. Compare magic
number to decide whether to clear entire BBRAM.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

brockus-zephyr commented on drivers/bbram/bbram_it8xxx2.c:
```diff
@@ -53,7 +59,30 @@ static const struct bbram_driver_api bbram_it8xxx2_driver_api = {
+	volatile uint8_t *bram_valid_flag0 = base_addr + BRAM_IDX_VALID_FLAGS0;
+	volatile uint8_t *bram_valid_flag1 = base_addr + BRAM_IDX_VALID_FLAGS1;
+	volatile uint8_t *bram_valid_flag2 = base_addr + BRAM_IDX_VALID_FLAGS2;
+	volatile uint8_t *bram_valid_flag3 = base_addr + BRAM_IDX_VALID_FLAGS3;
```
Why is volatile needed for these pointers?
    config->base_addr should not change and therefore removing optimization on base_addr for no reason.
    In this code, there is no redundant accesses to optimize out multiple reads that needs to be managed.

GTLin08 replied:
Done

brockus-zephyr commented on soc/riscv/riscv-ite/common/chip_chipregs.h:
```diff
@@ -1869,6 +1869,24 @@ struct gctrl_it8xxx2_regs {
+	BRAM_IDX_VALID_FLAGS2  = 0xbe,
+	BRAM_IDX_VALID_FLAGS3  = 0xbf
+
+};
```
I would prefer to remove the magic numbers in the code.

#define BRAM_MAGIC_FIELD_OFFSET 0xbc
enum bram_indices {
        /* This field is used to indicate BRAM is valid or not. */
        BRAM_IDX_VALID_FLAGS0  = BRAM_MAGIC_FIELD_OFFSET,
        BRAM_IDX_VALID_FLAGS1,
        BRAM_IDX_VALID_FLAGS2,
        BRAM_IDX_VALID_FLAGS3
};

GTLin08 replied:
Done.

GTLin08 committed:
ITE: cleanup: drivers/bbram: correct the default configuration format

1. Rename DT_COMPAT_ST_BBRAM_IT8XXX2 to COMAPT_ITE_IT8XXX2_BBRAM
2. The default configuration "$( )" should be a parentheses instead
   of curly brackets. The BBRAM for it8xxx2 can be driven after
   correcting.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

GTLin08 commented:
@brockus-zephyr add a new commit for cleanup in this PR.
